website/manual.md

```diff
@@ -23,68 +51,59 @@ program, it is runnable with nothing more than
+     or the network.
+3. Browser compatible: The subset of Deno programs which are written completely
+   in JavaScript and do not use the global `Deno` namespace (or feature test for
+   it), ought to also be able to be run in a modern web browser without change.
+4. Be able to serve HTTP efficiently
+5. Provide a great developer experience include built-in tooling.
+6. Does not leak browser or V8 concepts into user land.
+
+### Comparison to Node.js
+
+- Deno does not use `npm`
+  - It uses modules referenced as URLs or file paths
+- Deno does not use a `package.json`
+  - This is a non-goal. There are effective patterns [citation needed] for
+    managing dependencies.
+- Deno provides different APIs than node.
+- Deno requires explicit permissions for file, network, and environment access.
+  - Node is less secure out of the box.
+- Deno always dies on uncaught errors.
 - Uses "ES Modules" and does not support `require()`. Like the browser, allows
   imports from URLs:
 
-  ```ts
-  import * as log from "https://deno.land/std/log/mod.ts";
-  ```
+```ts
+import * as log from "https://deno.land/std/log/mod.ts";
+```
+
+### Other key behaviors
 
 - Remote code is fetched and cached on first execution, and never updated until
   the code is run with the `--reload` flag. (So, this will still work on an
   airplane. See `~/.deno/src` for details on the cache.)
-
-- File system and network access can be controlled in order to run sandboxed
-  code. Access between V8 (unprivileged) and Rust (privileged) is only done via
-  serialized messages. This makes it easy to audit. For example, to enable write
-  access use the flag `--allow-write` or for network access `--allow-net`.
-
-- Only ship a single executable.
-
-- Always dies on uncaught errors.
-
-- Browser compatible: The subset of Deno programs which are written completely
-  in JavaScript and do not use the global `Deno` namespace (or feature test for
-  it), ought to also be able to be run in a modern web browser without change.
-
-- [Aims to support top-level `await`.](https://github.com/denoland/deno/issues/471)
-
-- Be able to serve HTTP efficiently.
-  ([Currently it is relatively slow.](https://deno.land/benchmarks.html#req-per-sec))
-
-<!-- prettier-ignore-start -->
-<!-- see https://github.com/prettier/prettier/issues/3679 -->
-
-- Provide useful tooling out of the box:
-    - dependency inspector (`deno info`)
-    - code formatter (`deno fmt`),
-    - bundling (`deno bundle`)
-    - runtime type info (`deno types`)
-    - test runner (`deno test`)
-    - command-line debugger (`--debug`)
-      [not yet](https://github.com/denoland/deno/issues/1120)
-    - linter (`deno lint`) [not yet](https://github.com/denoland/deno/issues/1880)
-
-<!-- prettier-ignore-end -->
-
-### Non-goals
-
-- No `package.json`.
-
-- No npm.
-
-- Not explicitly compatible with Node.
+- Modules/files loaded from remote URLs are intended to be immutable and
```

Comment:
I wonder if this is actually a bug, and cache control should be respected? 😳

Reply:
I was talking with @kitsonk about this. I was saying that I think it _should_ be treated as immutable by default to encourage the practice in the community, but I think since putting a bit more thought to it, I think respecting cache headers would be fine. It doesn't prevent immutability from being possibly and a best practice in the community.

I think immutability in dependencies matters for a number of reasons, but especially because Deno is striving to be security minded.

One other approach we could use would be something like [subresource integrity](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity) where we could at least warn the user that a file has changed.

Regardless, I'll probably end up removing this bullet point from this PR and opening it in a separate issue for discussion.

Reply:
It looks like subresource integrity has actually been brought up before in https://github.com/denoland/deno/issues/200.